components/add_user_to_group_multiselect/add_user_to_group_multiselect.tsx

```diff
@@ -0,0 +1,271 @@
+                this.setUsersLoadingState(false);
+            });
+        } else {
+            this.props.actions.getProfiles().then(() => {
+                this.setUsersLoadingState(false);
+            });
+        }
+
+        this.props.actions.loadStatusesForProfilesList(this.props.profiles);
+    }
+
+    private handleDelete = (values: UserProfileValue[]): void => {
+        if (this.props.deleteUserCallback) {
+            this.props.deleteUserCallback(values);
+        }
+
+        this.setState({values});
+    };
+
+    private setUsersLoadingState = (loadingState: boolean): void => {
+        this.setState({
+            loadingUsers: loadingState,
+        });
+    };
+
+    private handlePageChange = (page: number, prevPage: number): void => {
+        if (page > prevPage) {
+            this.setUsersLoadingState(true);
+            if (this.props.groupId) {
+                this.props.actions.getProfilesNotInGroup(this.props.groupId, page + 1, USERS_PER_PAGE).then(() => {
+                    this.setUsersLoadingState(false);
+                });
+            } else {
+                this.props.actions.getProfiles(page + 1, USERS_PER_PAGE).then(() => {
+                    this.setUsersLoadingState(false);
+                });
+            }
+        }
+    };
+
+    public handleSubmit = (): void => {
+        this.setState({
+            saving: true,
+        });
+        const userIds = this.state.values.map((v) => v.id);
+        if (userIds.length === 0) {
+            return;
+        }
+        this.props.onSubmitCallback(this.state.values).then(() => {
+            this.setState({
+                saving: false,
+            });
+        });
+    };
+
+    public search = (searchTerm: string): void => {
+        const term = searchTerm.trim();
+        clearTimeout(this.searchTimeoutId);
+        this.setState({
+            term,
+        });
+
+        if (term) {
+            this.setUsersLoadingState(true);
+            this.searchTimeoutId = window.setTimeout(
+                async () => {
+                    await this.props.actions.searchProfiles(term, this.props.searchOptions);
+                    this.setUsersLoadingState(false);
+                },
+                Constants.SEARCH_TIMEOUT_MILLISECONDS,
+            );
+        }
+    };
+
+    private renderAriaLabel = (option: UserProfileValue): string => {
+        if (!option) {
+            return '';
+        }
+        return option.username;
+    }
+
+    renderOption = (option: UserProfileValue, isSelected: boolean, onAdd: (user: UserProfileValue) => void, onMouseMove: (user: UserProfileValue) => void) => {
+        let rowSelected = '';
+        if (isSelected) {
+            rowSelected = 'more-modal__row--selected';
+        }
+
+        return (
+            <div
+                key={option.id}
+                ref={isSelected ? this.selectedItemRef : option.id}
```

Comment:
I'm not 100% sure about this, I don't know if this ref is needed (looks like a leaky abstraction on the multiselect component), also the case where is not selected, why are we setting a ref equal to `option.id`, shouldn't it be something like `null` or `undefined`? (Anyway, this patterns looks like copied from the multiselect component, so I don't think this is an stopper.

Reply:
split this out into another component called `MultiSelectOption`, I removed the option.id as a ref and it's setup the same as `components/more_direct_channels/list_item/list_item.tsx`.